Applications/spire/source/charting/chart_view.cpp

```diff
@@ -43,23 +52,38 @@ ChartView::ChartView(ChartValue::Type x_axis_type, ChartValue::Type y_axis_type,
     ChartValue(current_time - boost::posix_time::hours(1)),
     ChartValue(Nexus::Money(10.10)));
   set_region(top_left, bottom_right);
+  setCursor(Qt::BlankCursor);
+  m_crosshair = imageFromSvg(":/icons/chart-cursor.svg", scale(16, 16));
+  m_dashed_line_pen.setDashPattern({static_cast<double>(scale_width(4)),
+    static_cast<double>(scale_width(4))});
 }
 
 ChartPoint ChartView::convert_pixels_to_chart(const QPoint& point) const {
-  return {};
+  return ChartPoint(
+    map_to(static_cast<double>(point.x()), 0.0,
+      static_cast<double>(m_x_origin), m_top_left.m_x, m_bottom_right.m_x),
+    map_to(static_cast<double>(point.y()), static_cast<double>(m_y_origin),
+      0.0, m_bottom_right.m_y, m_top_left.m_y));
 }
 
 QPoint ChartView::convert_chart_to_pixels(const ChartPoint& point) const {
-  return {};
+  return QPoint(
+    map_to(point.m_x, m_top_left.m_x, m_bottom_right.m_y, 0.0,
+      static_cast<double>(m_x_origin)),
+    map_to(point.m_y, m_bottom_right.m_y, m_top_left.m_y,
+      static_cast<double>(m_y_origin), 0.0));
 }
 
 void ChartView::set_crosshair(const ChartPoint& position) {
 }
 
 void ChartView::set_crosshair(const QPoint& position) {
+  m_crosshair_pos = position;
+  update();
 }
 
 void ChartView::reset_crosshair() {
+  m_crosshair_pos = QPoint();
```

Comment:
This is not correct and results in a subtle bug when the mouse is at position (0, 0). The mouse displays the crosshair cursor but no crosshair lines. You can pick another sentinel value to use or use an ```std::optional```.